### ISSUE TITLE
synth builds some ports which have "special" autoconf magic differently than natively, or those provided via FreeBSD repository

### ISSUE THREAD

**Issue by ArminGruner** [jrmarino/synth#63]
Hi (again :-),

I just stumbled over a very subtle problem after a system upgrade using a synth-built repository:

Eg. mail/procmail tries to detect with auto-configuration whether /usr/local/bin exists, and presets internal PATH defaults according to it.

If I use the "procmail" package from the FreeBSD repository, the man page (and the procmail binary) will list the default PATH variable containing "/usr/local/bin". 

If I use the "procmail" package which has been built with synth, both the man page, and the resulting procmail binary will not have the "/usr/local/bin" listed.

If I use ENTERAFTER= .. synth  test ..
and inspect the [pristine] build hierarchy, I indeed see no /usr/local/bin:

```
# ls -l /usr/local
total 0
drwxr-xr-x  4 root  wheel  256 Nov  4 14:07 etc
drwxr-xr-x  2 root  wheel   64 Nov  4 14:07 include
drwxr-xr-x  2 root  wheel  256 Nov  4 14:07 lib
drwxr-xr-x  3 root  wheel   64 Nov  4 14:07 libdata
drwxr-xr-x  5 root  wheel  192 Nov  4 14:07 man
drwxr-xr-x  2 root  wheel  192 Nov  4 14:07 sbin
drwxr-xr-x  5 root  wheel  192 Nov  4 14:07 share
```

The existing hierarchy seems to exist soleley because "pkg" has been installed initially.

I have been using poudriere before, and did not encounter differences between package buidling. So I assume it uses either some mtree pre-seeding or other kinds of building some /usr/local hierarchy.

**Comment by jrmarino** [comment]
that would be a problem with the port.
It shouldn't be testing for /usr/local/bin existence, it needs to use $PREFIX/bin in all cases.

In the clean environment, /usr/local/bin hasn't been created yet by the dependencies of procmail.

tldr;
The port needs to be fixed.

**Comment by jrmarino** [comment]
You should open a PR against the port, especially provide a patch if you can.  That way the patch can be committed in 2 weeks after sunpoet inevitably ignores the PR.

**Comment by ArminGruner** [comment]
I know it's not the fault of synth.

Having a clean build environment  (and later, a system which does not have build dependencies installed) is really a good thing [TM].

I just wanted to point out that I have been trapped by the syndrom described, and that other people are aware of it.

tldr;
Using a synth repository as an alternative should be done with care, because one doesn't know which ports don't behave "nicely" in a clean build environment.

**Comment by ArminGruner** [comment]
https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=214219

**Comment by jrmarino** [comment]
well, tools like synth and poudriere flush out the bad ports.  

In this case, the logic of procmail is wrong.  There's not law that the localbase has to be /usr/local so to assume that is wrong, it must be configured.

unfortunately there are so many ports that still make assumptions like this that testing for them hasn't even really started yet (poudriere has this ability but synth just uses /usr/local internally always).
